Sources/ConsoleKit/Utilities/GenerateAutocompleteCommand.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+
+struct GenerateAutocompleteCommand: Command {
+
+    var help: String { "Generate shell completion scripts for the executable" }
+
+    var rootCommand: AnyCommand?
+
+    init(rootCommand: AnyCommand? = nil) {
+        self.rootCommand = rootCommand
+    }
+
+    struct Signature: CommandSignature {
+
+        @Option(
+            name: "shell",
+            short: "s",
+            help: """
+            Generate completion script for SHELL [ \(Shell.allCases.map { "\($0)" }.joined(separator: " | ")) ].
+            Defaults to the "SHELL" environment variable if possible.
+            """,
+            completion: .values(of: Shell.self)
+        )
+        var shell: Shell?
+
+        @Option(
+            name: "output",
+            short: "o",
+            help: """
+            Write output to file at OUTPUT.
+            Defaults to printing to stdout.
+            """,
+            completion: .files()
+        )
+        var output: String?
+    }
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        guard let rootCommand = self.rootCommand else { fatalError("`rootCommand` was not initialized") }
+        guard let shell = signature.shell ?? self.environmentShell() else {
+            throw CommandError.missingRequiredArgument(signature.$shell.name)
+        }
+        // Reset the executable path
+        var context = context
+        context.input.executablePath = [context.input.executablePath.first!]
+        let script = rootCommand.renderCompletionScript(using: context, shell: shell)
+        if let output = signature.output, let contents = script.data(using: .utf8) {
+            try contents.write(to: URL(fileURLWithPath: output))
```

Comment:
Will this completely overwrite the file that is passed into the `output` option? If so, that should be mentioned in the `output` help message.

Reply:
Yeah, sorry, I meant to reply to your previous comment:

> From my understanding of auto-complete scripts, they get placed in the `.bashrc` or `.zshrc`

They need to be sourced or loaded in the `.bashrc` (or `.bash_profile`) or `.zshrc`, but the script's contents itself shouldn't really be appended to the file. Where exactly the script file should be saved really depends on a user's specific environment; I think the [Homebrew shell completion](https://docs.brew.sh/Shell-Completion) page says it best:

> since it is difficult to automatically configure `bash` and `zsh` completions in a robust manner, the Homebrew installer does not do it for you

I can add some instructions to the doc comments similar to [these SPM instructions](https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md#using-shell-completion-scripts).

Reply:
It's common for command-line tools to output instructions for things that need to be added to `.bashrc` or `.zshrc`. We could do something like that, e.g.:

```
$ <executable> autocomplete --shell bash --output <file>

bash completion script written to <file>

Add the following to your ~/.bashrc or ~/.bash_profile to enable autocompletion:

[ -f "<file>" ] && . "<file>"

```

Reply:
Yeah, that sounds like a good idea.

Reply:
👍 I also added a `--quiet` flag so this informational output can be suppressed if desired.